Content.Goobstation.Shared/MartialArts/SharedMartialArtsSystem.SpaceBear.cs

```diff
@@ -0,0 +1,126 @@
+        SubscribeLocalEvent<GrantSpaceBearComponent, ExaminedEvent>(OnGrantCQCExamine);
+
+    }
+
+    #region Generic Methods
+
+    private void OnGrantCQCUse(EntityUid ent, GrantSpaceBearComponent comp, UseInHandEvent args)
+    {
+        if (!_netManager.IsServer)
+            return;
+
+        if (comp.Used)
+        {
+            _popupSystem.PopupEntity(Loc.GetString(comp.LearnFailMessage, ("manual", Identity.Entity(ent, EntityManager))),
+                args.User,
+                args.User);
+            return;
+        }
+
+        if (!TryGrantMartialArt(args.User, comp))
+            return;
+        _popupSystem.PopupEntity(Loc.GetString(comp.LearnMessage), args.User, args.User);
+        comp.Used = true;
+        if (!_entityManager.HasComponent<FireProtectionComponent>(ent))
+        {
+            _entityManager.AddComponent<FireProtectionComponent>(ent);
+        }
+        var fireProt = _entityManager.GetComponent<FireProtectionComponent>(ent);
+        _entityManager.AddComponent<StaminaDamageResistanceComponent>(ent);
+        var stamProt = _entityManager.GetComponent<StaminaDamageResistanceComponent>(ent);
+        stamProt.Coefficient = 0.8f; // 20% stamina damage resistance
+        //_tag.AddTag(ent, "Bear");
+        _fireProtectionSystem.OnSetComponent(fireProt, 0.0f, "spacebear-description-fireimmunity"); // Doesn't work?
+    }
+
+    private void OnSpaceBearAttackPerformed(Entity<MartialArtsKnowledgeComponent> ent, ref ComboAttackPerformedEvent args)
+    {
+        if (args.Weapon != args.Performer || args.Target == args.Performer)
+            return;
+
+        switch (args.Type)
+        {
+            case ComboAttackType.Harm:
+                DoDamage(ent, args.Target, "Slash", 10, out _);
+                _stamina.TakeStaminaDamage(args.Performer, -20f, applyResistances: false);
+                break;
+        }
+    }
+    #endregion
+
+    #region Combo Methods
+
+    private void OnBearJaws(Entity<CanPerformComboComponent> ent, ref BearJawsPerformedEvent args)
+    {
+        if (!_proto.TryIndex(ent.Comp.BeingPerformed, out var proto)
+            || !TryUseMartialArt(ent, proto, out var target, out var downed))
+            return;
+
+        if (downed)
+        {
+            DoDamage(ent, target, "Slash", 20, out _);
+            _stamina.TakeStaminaDamage(ent, -20f, applyResistances: false);
+        }
+        else
+        {
+            DoDamage(ent, target, "Slash", 10, out _);
+        }
+        ComboPopup(ent, target, proto.Name);
+    }
+
+    private void OnPawSlam(Entity<CanPerformComboComponent> ent, ref PawSlamPerformedEvent args)
+    {
+        if (!_proto.TryIndex(ent.Comp.BeingPerformed, out var proto)
+            || !TryUseMartialArt(ent, proto, out var target, out var downed))
+            return;
+        if (downed)
+        {
+            DoDamage(ent, target, "Slash", 10, out _);
+            _stun.TryKnockdown(target, TimeSpan.FromSeconds(4), true);
```

Comment:
You are knocking down already knocked down people (knockdown lock), is it intended?